src/generators/rust/renderers/EnumRenderer.ts

```diff
@@ -43,26 +43,47 @@ ${additionalContent}`;
   runStructMacroPreset(): Promise<string> {
     return this.runPreset('structMacro');
   }
+
+  /**
+   * Returns the type for the JSON value
+   */
+  renderEnumValueType(value: any): string {
+    switch (typeof value) {
+    case 'boolean':
+      return 'bool';
+    case 'bigint':
+      return 'i64';
+    case 'number': {
+      return 'f64';
+    }
+    case 'object': {
+      return 'HashMap<String, serde_json::Value>';
+    }
+    case 'string':
+    default: {
+      return 'String';
+    }
+    }
+  }
 }
 
 export const RUST_DEFAULT_ENUM_PRESET: EnumPresetType<RustOptions> = {
   self({ renderer }) {
     return renderer.defaultSelf();
   },
-  item({ item }) {
-    if (item.value === 'HashMap<String, serde_json::Value>') {
-      return `${item.key}(${item.value})`;
+  item({ item, renderer }) {
+    const typeOfEnumValue = renderer.renderEnumValueType(item.value);
+    if (typeOfEnumValue === 'HashMap<String, serde_json::Value>') {
+      return `${item.key}(${typeOfEnumValue})`;
     }
     return `${item.key}`;
   },
-  itemMacro({ itemIndex, model }) {
-    const originalInput = model.originalInput.enum[Number(itemIndex)];
+  itemMacro({ item }) {
     const serdeArgs = [];
-    if (typeof originalInput === 'object') {
+    if (typeof item.value === 'object') {
       serdeArgs.push('flatten');
     } else {
-      const rename = model.originalInput.enum[Number(itemIndex)].toString();
-      serdeArgs.push(`rename="${rename}"`);
+      serdeArgs.push(`rename="${item.value}"`);
```

Comment:
`rename` is a serde macro that will munge the property name on serialization/deserialization, for example munging camelCase to snake_case.

Is `${item.value}` the right value here? Usually, rename is used to rewrite the property name. Quick Example :point_down: 

https://serde.rs/field-attrs.html#field-attributes

```rust

// ExampleStruct will serialize to/from camelCased JSON: `{ "propertyOne": 10 }`
#[derive(Debug, Clone, Deserialize, Serialize)]
struct ExampleStruct {
    #[serde(rename = "propertyOne")]
    property_one: i32
}
```

Reply:
It does because the enum value constrainer for Rust returns the JSON value as is now, i.e. the same you did with `model.originalInput.enum[Number(itemIndex)].toString();` 🙂 

All the tests for the enum generator are exactly the same snapshot.